gateway/src/apicast/policy/standalone/standalone.lua

```diff
@@ -0,0 +1,364 @@
+-- This is a standalone description.
+
+local Policy = require('apicast.policy')
+local PolicyChain = require('apicast.policy_chain')
+local Upstream = require('apicast.upstream')
+local Configuration = require('apicast.policy.standalone.configuration')
+local resty_env = require('resty.env')
+local _M = Policy.new('standalone')
+local tab_new = require('resty.core.base').new_tab
+local insert = table.insert
+local pairs = pairs
+local assert = assert
+local format = string.format
+local concat = table.concat
+
+local setmetatable = setmetatable
+
+local function load_configuration(url)
+  local configuration, err = Configuration.new(url)
+
+  if configuration then
+    configuration, err = configuration:load()
+  end
+
+  return configuration, err
+end
+
+local new = _M.new
+--- Initialize the Standalone APIcast policy
+-- @tparam[opt] table config Policy configuration.
+function _M.new(config)
+  local self = new(config)
+
+  self.url = (config and config.url) or resty_env.get('APICAST_CONFIGURATION')
+
+  return self
+end
+
+
+local empty_chain = PolicyChain.new()
+
+-- forward all policy request methods to the policy chain
+for _,phase in Policy.request_phases() do
+  _M[phase] = function(self, context, ...)
+    if context[self] then
+      return context[self][phase](context[self], context, ...)
+    end
+  end
+end
+
+local function build_objects(constructor, list)
+  if not list then return nil end
+
+  local objects = tab_new(#list, 0)
+
+  for i=1, #list do
+    local object = constructor(list[i])
+
+    if object.name then
+      objects[object.name] = object
+    end
+
+    objects[i] = object
+  end
+
+  return objects
+end
+
+local Route = { }
+
+local function build_routes(configuration)
+  return build_objects(Route.new, configuration.routes)
+end
+
+local Destination = { }
+do
+  local Destination_mt = { __index = Destination }
+
+  function Destination.new(config)
+    if not config then return nil end
+
+    return setmetatable({
+      service = config.service,
+      http_response = config.http_response,
+    }, Destination_mt)
+  end
+end
+
+local Condition = { }
+do
+
+  local Condition_mt = { __index = Condition }
+
+  local operations = {
```

Comment:
In a future iteration, we should think about how to reuse what we already did for the Conditional policy or add there whatever is missing to support what you introduced here.

Reply:
Agreed. Wanted to have this here, for now, to see how it evolves. But in the future there should be just one way to do this.